### ISSUE TITLE
How can I use it with the latest Ionic2 version?

### ISSUE THREAD

**Issue by OliverTr54** [twinssbc/Ionic2-Calendar#11]
Hey folks,

I really have to have this great plugin, but how can I use it with the latest Ionic2 version. Here are some more information about my used enviroment:

```
Ionic Framework Version: 2.0.0-rc.0
Ionic CLI Version: 2.1.2
Ionic App Lib Version: 2.1.1
Ionic App Scripts Version: 0.0.27
OS: Distributor ID:     Ubuntu Description:     Ubuntu 16.04.1 LTS
Node Version: v4.2.6

```

So actually what I did - among your documentation I included it differently, because I was not able to include it like you described it on your Github page neither as described in the README file.

I did the following (File: app.module.ts):

```
...
import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { CalendarComponent } from 'ionic2-calendar/ionic2-calendar'
...
@NgModule({
  declarations: [
    MyApp,
    HomePage,
    Calendar,
    CalendarComponent
  ],
...
  schemas: [CUSTOM_ELEMENTS_SCHEMA]
})

```

When I build or serve it I don't get any mistake, but when I try to open it, e.g. in the browser I constantly get this message in the console.

Unhandled Promise rejection: Template parse errors:

```
Can't bind to 'formatDay' since it isn't a known property of 'monthview'. ("ngSwitch]="calendarMode" class="calendar-container">
            <monthview *ngSwitchCase="'month'" [ERROR ->][formatDay]="formatDay" [formatDayHeader]="formatDayHeader" [formatMonthTitle]="formatMonthTitle"
   "): CalendarComponent@2:47
Can't bind to 'formatDayHeader' since it isn't a known property of 'monthview'. (" class="calendar-container">
            <monthview *ngSwitchCase="'month'" [formatDay]="formatDay" [ERROR ->][formatDayHeader]="formatDayHeader" [formatMonthTitle]="formatMonthTitle"
             [startingDayMo"): CalendarComponent@2:71
Can't bind to 'formatMonthTitle' since it isn't a known property of 'monthview'. ("     <monthview *ngSwitchCase="'month'" [formatDay]="formatDay" [formatDayHeader]="formatDayHeader" [ERROR ->][formatMonthTitle]="formatMonthTitle"
             [startingDayMonth]="startingDayMonth" [showEventDe"): CalendarComponent@2:107
Can't bind to 'startingDayMonth' since it isn't a known property of 'monthview'. ("="formatDay" [formatDayHeader]="formatDayHeader" [formatMonthTitle]="formatMonthTitle"
             [ERROR ->][startingDayMonth]="startingDayMonth" [showEventDetail]="showEventDetail" [noEventsLabel]="noEventsLa"): CalendarComponent@3:13
Can't bind to 'showEventDetail' since it isn't a known property of 'monthview'. ("DayHeader" [formatMonthTitle]="formatMonthTitle"
             [startingDayMonth]="startingDayMonth" [ERROR ->][showEventDetail]="showEventDetail" [noEventsLabel]="noEventsLabel" [eventSource]="eventSource"
...
```

and so on...

So what can I do? I need this plugin urgently, because it's excactly what I was looking for.

Thanks,
Oliver

**Comment by twinssbc** [comment]
@OliverTr54 I have just upgraded the code to use the latest Ionic2 version, which is 2.0.0.rc.1.
But I haven't uploaded it to NPM, because some slight rendering issue.
So if you follow the README to npm install it, you still get the previous version, that's why you got this error.
I will upload the latest version to NPM this weekend.

**Comment by twinssbc** [comment]
@OliverTr54 I have published version 0.0.8 which depends on Angular 2.0.0 and Ionic 2.0.0.rc.1.
Feel free to have a try. Thanks.

**Comment by RyanTLX** [comment]
Hi @OliverTr54 , how did you get the calendar to work? @twinssbc 
I'm trying this calendar out from a blank ionic 2 project.

My ionic version
```
Cordova CLI: 6.4.0
Ionic Framework Version: 2.0.0-rc.4
Ionic CLI Version: 2.1.14
Ionic App Lib Version: 2.1.7
Ionic App Scripts Version: 0.0.47
ios-deploy version: 1.9.0
ios-sim version: 5.0.12
OS: macOS Sierra
Node Version: v6.9.2
Xcode version: Xcode 8.2.1 Build version 8C1002
```

My app.module.ts
```
import { NgModule, ErrorHandler } from '@angular/core';
import { IonicApp, IonicModule, IonicErrorHandler } from 'ionic-angular';
import { MyApp } from './app.component';
import { HomePage } from '../pages/home/home';

import { NgCalendarModule  } from 'ionic2-calendar';

@NgModule({
  declarations: [
    MyApp,
    HomePage
  ],
  imports: [
    NgCalendarModule,
    IonicModule.forRoot(MyApp)
  ],
  bootstrap: [IonicApp],
  entryComponents: [
    MyApp,
    HomePage
  ],
  providers: [{provide: ErrorHandler, useClass: IonicErrorHandler}]
})
export class AppModule {}
```

Then in home.html I inserted the calendar code as such
```
<ion-header>
  <ion-navbar>
    <ion-title>
      Home
    </ion-title>
  </ion-navbar>
</ion-header>

<ion-content padding>
    <calendar [eventSource]="eventSource"
        [calendarMode]="calendar.mode"
        [currentDate]="calendar.currentDate"
        (onCurrectDateChanged)="onCurrentDateChanged($event)"
        (onRangeChanged)="reloadSource(startTime, endTime)"
        (onEventSelected)="onEventSelected($event)"
        (onTitleChanged)="onViewTitleChanged($event)"
        (onTimeSelected)="onTimeSelected($event)"
        step="30">
      </calendar>
</ion-content>
```

But I'm getting the following error
<img width="812" alt="screen shot 2016-12-28 at 9 32 41 pm" src="https://cloud.githubusercontent.com/assets/6291743/21523009/4e284e0e-cd45-11e6-9b70-68eb5ec524c8.png">

Please pardon my ignorance. Just getting started with the ionic framework.
